node/pkg/governor/governor.go

```diff
@@ -611,31 +656,35 @@ func (gov *ChainGovernor) ProcessMsgForTime(msg *common.MessagePublication, now
 	// - This will cause the summed value of Sui to decrease.
 	emitterChainEntry.transfers = append(emitterChainEntry.transfers, transfer)
 
-	// Add inverse transfer to destination chain entry if this asset can cancel flows.
-	key := tokenKey{chain: token.token.chain, addr: token.token.addr}
-
-	tokenEntry := gov.tokens[key]
-	if tokenEntry != nil {
-		// Mandatory check to ensure that the token should be able to reduce the Governor limit.
-		if tokenEntry.flowCancels {
-			if destinationChainEntry, ok := gov.chains[payload.TargetChain]; ok {
-				if err := destinationChainEntry.addFlowCancelTransferFromDbTransfer(&dbTransfer); err != nil {
-					return false, err
-				}
-			} else {
-				gov.logger.Warn("tried to cancel flow but chain entry for target chain does not exist",
-					zap.String("msgID", msg.MessageIDString()),
-					zap.String("hash", hash), zap.Error(err),
-					zap.Stringer("target chain", payload.TargetChain),
-				)
-			}
+	if gov.flowCancelEnabled {
+		_, err = gov.tryAddFlowCancelTransfer(&transfer)
+		if err != nil {
+			gov.logger.Warn("Error when attempting to add a flow cancel transfer",
+				zap.Error(err),
+			)
+			return false, err
```

Comment:
Same question for this early return, do we want this? And should the `newTransferFromDbTransfer` come before the `StoreTransfer` call?

Reply:
I believe this is done now.